R/lgcm.R

```diff
@@ -0,0 +1,793 @@
+  }
+  
+  # Add regressions
+  Reg <- if (length(options[["regressions"]]) > 0)
+    paste0("\n\n# Regressions\n", paste(curve, collapse = " + "), " ~ ", 
+           paste(.v(options[["regressions"]]), collapse = " + "))
+  else NULL
+  
+  # Add dummy variables
+  Dum <- if (length(options[["dummy_names"]]) > 0)
+    paste0("\n\n# Dummy-coded categorical predictors\n", paste(curve, collapse = " + "), " ~ ", 
+           paste(.v(options[["dummy_names"]]), collapse = " + "))
+  
+  # Add time-varying covariates
+  # eww this is hard
+  
+  # Put everything together
+  paste0(Hed, LGC, Cov, Reg, Dum)
+}
+
+# Output functions ----
+.lgcmModelContainer <- function(jaspResults, options) {
+  if (!is.null(jaspResults[["modelContainer"]])) {
+    modelContainer <- jaspResults[["modelContainer"]]
+  } else {
+    modelContainer <- createJaspContainer()
+    modelContainer$dependOn(c(
+      "variables", "regressions", "covariates", "categorical", "timings", 
+      "intercept", "linear", "quadratic", "cubic", "covar",
+      "se", "bootstrapNumber"
+    ))
+    jaspResults[["modelContainer"]] <- modelContainer
+  }
+  
+  return(modelContainer)
+}
+
+.lgcmFitTable <- function(modelContainer, dataset, options, ready) {
+  if (!is.null(modelContainer[["maintab"]])) return()
+  maintab <- createJaspTable("Chi-square Test")
+  maintab$addColumnInfo(name = "mod",    title = "Model",        type = "string")
+  maintab$addColumnInfo(name = "chisq",  title = "\u03a7\u00b2", type = "number", format = "dp:3")
+  maintab$addColumnInfo(name = "df",     title = "df",           type = "integer")
+  maintab$addColumnInfo(name = "pvalue", title = "p",            type = "number", format = "dp:3;p:.001")
+  
+  modelContainer[["maintab"]] <- createJaspContainer("Model fit")
+  modelContainer[["maintab"]]$position <- 1
+  modelContainer[["maintab"]][["chisqtab"]] <- maintab
+  
+  # add data to the table!
+  if (!ready) return()
+  lgcmResult <- .lgcmComputeResults(modelContainer, dataset, options)
+  if (modelContainer$getError()) return()
+  
+  fm <- lavaan::fitMeasures(lgcmResult)
+  maintab[["mod"]]    <- c("Baseline model", "Growth curve model")
+  maintab[["chisq"]]  <- fm[c("baseline.chisq", "chisq")]
+  maintab[["df"]]     <- fm[c("baseline.df", "df")]
+  maintab[["pvalue"]] <- c(NA, fm["pvalue"])
+  
+  # display warnings in footnote
+  
+  admissible <- JASP:::.withWarnings(lavaan:::lav_object_post_check(lgcmResult))
+  
+  if (!admissible$value) {
+    maintab$addFootnote(
+      message = gettextf(
+        "The model is not admissible: %s",
+        .decodeVarsInMessage(names(dataset), admissible$warnings[[1]]$message)
+      ),
+      symbol = gettext("Warning.")
+    )
+  }
+  
+}
+
+.lgcmParameterTables <- function(modelContainer, dataset, options, ready) {
+  partabs <- if (!is.null(modelContainer[["partabs"]])) {
+    modelContainer[["partabs"]]
+  } else {
+    modelContainer[["partabs"]] <- createJaspContainer("Parameter estimates")
```

Comment:
missing `gettext()`

Reply:
these seem to be generally missing in this file

Reply:
oopsie